### ISSUE TITLE
batched_mixed.Ewald: non-PBC potentials halved (factor-of-2 in calculators.py:71)

### ISSUE THREAD

**Issue by sirmarcel** [lab-cosmo/jax-pme#25]
> *Filed by Claude Code on behalf of @sirmarcel (he asked me to write this up after we tracked the bug down together).*

## Summary

`jaxpme.batched_mixed.Ewald` returns non-PBC potentials that are **exactly half** the physically correct value. Forces pass because the factor-of-2 scales out through gradients, so this is invisible to most MD/force tests — but any downstream code that consumes the bare potentials (e.g. as features in an MLIP) will be off by a constant.

## Reproducer

```python
import numpy as np, jax.numpy as jnp
from ase.build import molecule
from jaxpme.batched_mixed import Ewald
from jaxpme.batched_mixed.batching import prepare, get_batch

atoms = molecule("H2O")              # any non-PBC structure
atoms.set_pbc(False)
N = len(atoms)
q = np.random.default_rng(0).normal(size=N)

# direct-sum Coulomb, pair-wise over all N*N
r = np.linalg.norm(atoms.positions[None, :] - atoms.positions[:, None], axis=-1)
diag = np.eye(N, dtype=bool)
inv_r = np.where(diag, 0.0, 1.0 / np.where(diag, 1.0, r))
reference = inv_r @ q                 # V_i = sum_{j != i} q_j / r_ij

calc = Ewald()
structure = prepare(atoms, cutoff=5.0)
q_padded, batch, nopbc, pbc = get_batch([structure])
q_in = np.zeros(q_padded.shape[0]); q_in[:N] = q
observed = np.asarray(calc.potentials(jnp.array(q_in), batch, nopbc, pbc))[:N]

print("ratio:", observed / reference)
# -> [0.5, 0.5, 0.5]
```

Run on `jax-pme` at `ebf8d8b4a1704922cf467077e6a3b03d1f912262`:

```
ratio: [0.50000007 0.50000004 0.50000026]
```

## Root cause

`src/jaxpme/batched_mixed/batching.py:319` generates the non-PBC pair list as `np.triu_indices(N, k=1)` — each undirected pair *once* (upper triangle only, no self-pairs).

`src/jaxpme/batched_mixed/calculators.py:59-71` then accumulates using the "full neighbor list" convention (one `segment_sum` scattered to `centers`, another to `others`) and divides by 2:

```python
real_space = jax.ops.segment_sum(
    charges[batch_nopbc.others] * pot.real(extra_r),
    batch_nopbc.centers,
    num_segments=N_all,
)
real_space += jax.ops.segment_sum(
    charges[batch_nopbc.centers] * pot.real(extra_r),
    batch_nopbc.others,
    num_segments=N_all,
)
real_space = (real_space * batch.atom_mask * ~pbc_mask) / 2   # <-- bug
```

The `/ 2` is only correct if each pair appears twice in the list (both (i,j) and (j,i)). With triu-only input, the two `segment_sum`s already produce the symmetric potential. Dividing by 2 halves the correct answer.

## Fix

One-line change: drop the `/ 2` on the last line above.

## Why tests didn't catch it

`tests/test_batched_mixed_calculator.py::test_mixed` is the one test that directly checks non-PBC potential values, and **its reference has the same `/ 2` built in** (line 70):

```python
all_distances = atoms2.get_all_distances()
mask = all_distances != 0.0
one_over_r = np.where(mask, 1 / np.where(mask, all_distances, 1.0), 0.0)
bare_potentials = np.einsum("ij,j->ij", one_over_r, charges)
reference_potentials = bare_potentials.sum(axis=1) / 2   # <-- same /2
```

So the test is self-consistent with the buggy code, not with physics.

`test_batched_mixed_equivalence` above it (line 34-53) prepares a non-PBC structure (`atoms_no_pbc`) through the batched calculator and would catch this against the serial `jaxpme.Ewald` reference — but the assertion loop iterates only over `structures` (the original PBC ones) and never indexes into the non-PBC entry. The non-PBC path is exercised but never asserted on.

The H₂ smoke test (`test_mixed_pbc`, line 136) just asserts `not NaN` and `energy < 0` for an opposite-charge pair, which passes for both correct (−1 eV) and halved (−0.5 eV) potentials.

## Suggested fix to tests

1. In `test_mixed`, drop the `/ 2` from `reference_potentials` (line 70) so the test validates against physical values rather than the buggy code.
2. In `test_batched_mixed_equivalence`, extend the assertion loop to also compare the non-PBC structure against serial `Ewald` with `pbc=False`.

**Comment by sirmarcel** [comment]
This is not a bug, it is intended behaviour. See #27 . Closing.

**Comment by sirmarcel** [comment]
Correction / retraction: after discussion with @sirmarcel and the jax-pme developers, the `/ 2` in `batched_mixed.Ewald`'s non-PBC branch is actually correct — it's consistent with the convention the PBC branch uses (the returned potential is the one where `E = sum_i q_i V_i` recovers the pairwise energy directly, without an outer factor of 1/2).

The factor-of-2 discrepancy I observed in the original report is real, but the "wrong side" of it is the consumer of this library, not this library itself: an older hand-rolled `sum_{j != i} q_j / r_ij` path in a downstream MLIP project used the physicist's "sum-of-pair-contributions" convention without the outer `1/2`, which disagreed with jax-pme. The fix belongs on the consumer side, not here.

Please close this issue — sorry for the noise. The tests as written are fine and correctly encode the canonical convention.
